### DISCUSSION TITLE
Code Flow

### DISCUSSION THREAD

**Discussion by AmanBhanse** [edvilme/animated-web-components#6] [Q&A]
@edvilme 

So I want to start the discussion about how the basic code works?

suppose i have below code
```
class RisingElement extends AnimationElement{
   
    constructor(){
        super();
        this.attachShadow({mode: 'open'});

    }

    connectedCallback(){
        this.render();
    }
    render(){
        console.log(" I am inside Render");
    }
    animate(){
       console.log("I Am animating");
    }
   
}
customElements.define('rising-element', RisingElement);
```

When i use this directive then I see animate() is not called.
So how the code flows?

**Comment by edvilme** [comment]
Oh I see. I think it is not properly configured (I was --and still am-- relatively new to WebComponents at the time. You have to use an attribute to get it started. Will check and refactor later. Have a look at AnimationElement.
Tomorrow I will do some refactoring to make it easier to extend and explain. 

Thanks A LOT for your enthusiasm and support!
